mm2src/coins/eth/web3_transport/mod.rs

```diff
@@ -134,26 +134,46 @@ pub struct FeeHistoryResult {
     pub priority_rewards: Option<Vec<Vec<U256>>>,
 }
 
-/// Generates a signed message and inserts it into the request payload.
+/// Generates a Quicknode payload JSON string by inserting a signed message into the request payload.
 pub(super) fn handle_quicknode_payload(
     proxy_auth_validation_generator: &Option<ProxyAuthValidationGenerator>,
     request: &Call,
 ) -> Result<String, Web3RpcError> {
+    let signed_message = generate_signed_message(proxy_auth_validation_generator)?;
+
+    let auth_request = QuicknodePayload {
+        request,
+        signed_message,
+    };
+
+    Ok(to_string(&auth_request))
+}
+
+/// Generates a signed message JSON string if proxy authentication is enabled.
+pub(crate) fn generate_auth_header(
+    proxy_auth_validation_generator: &Option<ProxyAuthValidationGenerator>,
+    gui_auth: bool,
+) -> Result<Option<String>, Web3RpcError> {
+    if !gui_auth {
+        return Ok(None);
+    }
+    let signed_message = generate_signed_message(proxy_auth_validation_generator)?;
+    Ok(Some(serde_json::to_string(&signed_message)?))
+}
+
+/// Generates a signed message using the provided ProxyAuthValidationGenerator
+fn generate_signed_message(
+    proxy_auth_validation_generator: &Option<ProxyAuthValidationGenerator>,
+) -> Result<KomodefiProxyAuthValidation, Web3RpcError> {
     let generator = proxy_auth_validation_generator
         .clone()
-        .ok_or_else(|| Web3RpcError::Internal("ProxyAuthValidationGenerator is not provided for".to_string()))?;
+        .ok_or_else(|| Web3RpcError::Internal("ProxyAuthValidationGenerator is not provided".to_string()))?;
```

Comment:
is this an internal error?

Reply:
Yeah, I believe its correct error type, as it is mm2 task to build and handle auth generator/message properly
https://github.com/KomodoPlatform/komodo-defi-framework/pull/2100/files#diff-a98662dd9da4f8f899ca3eb5f35e4fdb82d4d594ff43bf683e52d860185b0ef0L138-L161

Additional info about naming.
In the dev branch, types for the proxy layer use names like `GuiAuth..smth else`, but they should actually be renamed to `ProxyAuth...`, `proxy_payload`, etc. Additionally, the payload type for the Quicknode service should maintain its naming as `quicknode_payload` to indicate that this proxy data will target Quicknode endpoints. Note: Since the Moralis API accepts HTTP GET requests, we don't have a payload for it. we only provide an auth signed message in the Header for validation purposes.